add documents for open-sourcing

tommyknows opened the pull request:
**chore: Add security document**

According to the open-sourcing notion page.



**chore: add support notice**

TODO: do we actually want to do this through support, or do we also want
to use github issues with that? I'd vote for a "proper" OS-workflow.



**chore: require secretName in Helm Chart**

This commit makes the `secretName` value for the Helm chart installation
`required`, meaning that Helm will error out if the value is not
provided.



**chore: improve Helm values.yaml descriptions**




**docs: improve README**

As part of TIKI-61.



**docs: add CONTRIBUTING document**




**docs: add code of conduct**

According to [TIKI-64](https://snyksec.atlassian.net/browse/TIKI-64).


[TIKI-64]: https://snyksec.atlassian.net/browse/TIKI-64?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

Commit:
chore: add security document

according to the open-sourcing notion page.

Commit:
chore: add support notice

TODO: do we actually want to do this through support, or do we also want
to use github issues with that? I'd vote for a "proper" OS-workflow.

Commit:
chore: require secretName in Helm Chart

This commit makes the `secretName` value for the Helm chart installation
`required`, meaning that Helm will error out if the value is not
provided.

Commit:
chore: improve Helm values.yaml descriptions

Commit:
docs: improve README

As part of TIKI-61.

Commit:
docs: add CONTRIBUTING document

Commit:
docs: add code of conduct

according to [TIKI-64](https://snyksec.atlassian.net/browse/TIKI-64).

Commit:
docs: add architecture document

craigfurman reviewed (changes requested):
👍  this is very nearly there, good work! Just a few notes.

Commit:
chore: add NOTICE ad bundle licenses and ensure they're up-to-date

this commit adds all licenses from our dependencies, which has been done
with the `go-licenses` tool. Additionally, it adds a `make
generate-licenses` step to easily re-run this command and regenerate
that folder.

The `NOTICE` file simply links to that folder.

Also, a CI check ensures that this folder is always up-to-date.

craigfurman reviewed (approved):
🚀
